### PR TITLE
feat: tcpinfo on peers

### PR DESCRIPTION
**Pull request by wolf31o2**
Support TCP connection to peers to get round trip times.

- Import `tcp` and `tcpinfo`
- Import `chainsync`
- Create `tcpinfoRtt()` to return round trip times in milliseconds
- Incorporate feedback from @agaffney on simplifying loop expressions
- Disable verbose debugging output on peers page
- Move Prometheus related items to prometheus.go
- Create `createRemoteClientConnection()` to connect to a remote host over TCP
- Add (currently unused) `getRemoteTip()` to get a Tip via Ouroboros NtN

image

### CODE REVIEW

File: node.go [hunk 1]
```diff
@@ -109,9 +117,47 @@ func getProtocolParams(cfg *Config) *localstatequery.CurrentProtocolParamsResult
 	if err != nil {
 		return result
 	}
-	// Start our client
-	o.LocalStateQuery().Client.Start()
-	result, err = o.LocalStateQuery().Client.GetCurrentProtocolParams()
+	// Query our client
+	oConn.LocalStateQuery().Client.Start()
```
Comment (agaffney) [inline node.go:121]: You could drop this entirely now

File: node.go [hunk 2]
```diff
@@ -73,16 +80,16 @@ func getGenesisConfig(cfg *Config) *localstatequery.GenesisConfigResult {
 	if err != nil {
 		return result
 	}
-	// Start our client
-	o.LocalStateQuery().Client.Start()
-	result, err = o.LocalStateQuery().Client.GetGenesisConfig()
+	// Query our client
+	oConn.LocalStateQuery().Client.Start()
```
Comment (agaffney) [inline node.go:84]: You could drop this entirely now

File: node.go [hunk 3]
```diff
@@ -109,9 +117,47 @@ func getProtocolParams(cfg *Config) *localstatequery.CurrentProtocolParamsResult
...
+		ouroboros.WithErrorChan(errorChan),
+		ouroboros.WithNodeToNode(true),
+		ouroboros.WithKeepAlive(false),
+		ouroboros.WithChainSyncConfig(buildChainSyncConfig()),
+	)
+	if err != nil {
+		return result
+	}
+	// Query our client
+	oConn.ChainSync().Client.Start()
```
Comment (agaffney) [inline node.go:159]: You can drop the explicit start now

File: node.go [hunk 4]
```diff
@@ -109,9 +117,47 @@ func getProtocolParams(cfg *Config) *localstatequery.CurrentProtocolParamsResult
...
+			os.Exit(1)
+		}
+	}()
+	oConn, err := ouroboros.NewConnection(
+		ouroboros.WithConnection(conn),
+		ouroboros.WithNetworkMagic(uint32(cfg.Node.NetworkMagic)),
+		ouroboros.WithErrorChan(errorChan),
+		ouroboros.WithNodeToNode(true),
+		ouroboros.WithKeepAlive(false),
+		ouroboros.WithChainSyncConfig(buildChainSyncConfig()),
```
Comment (agaffney) [inline node.go:153]: It may still work without passing this at all if you're just getting the current tip

**Review by agaffney — APPROVED** [review]

File: node.go (hunk 1 above)
Comment (wolf31o2) [reply node.go:121]: No. I need to upgrade gOuroboros for that and I'm avoiding that for the moment to keep the current version compatible with 1.35.7 and will upgrade that at a later date.

File: node.go (hunk 2 above)
Comment (wolf31o2) [reply node.go:84]: Not until I upgrade gOurorboros.

File: node.go (hunk 3 above)
Comment (wolf31o2) [reply node.go:159]: Not until I upgrade gOurorboros.

File: node.go (hunk 4 above)
Comment (wolf31o2) [reply node.go:153]: I haven't tested that, yet. I'll let you know.

**Review by agaffney — APPROVED** [review]

**Pull request merged**